### ISSUE TITLE
Exception on master when an exception is thrown on minion

### ISSUE THREAD

**Issue by yannis666** [saltstack/salt#26040]
I (purposefully) invoked a custom plugin which raises an exception. This causes a failure on the master in fetching error results.
```console
[root@flt-master /]# salt-master --version
salt-master 2015.5.3 (Lithium)
```

On master I see:
```python-traceback
[root@flt-master /]# salt 'pod-master-aus-jsf' oracle_lcm_orchestrator.start_immediate failure_raise raise
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
KeyError: 'id'
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 349, in salt_main
    client.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/salt.py", line 195, in run
    for full_ret in cmd_func(**kwargs):
  File "/usr/lib/python2.6/site-packages/salt/client/__init__.py", line 595, in cmd_cli
    **kwargs):
  File "/usr/lib/python2.6/site-packages/salt/client/__init__.py", line 1297, in get_cli_event_returns
    expect_minions=(verbose or show_timeout)
  File "/usr/lib/python2.6/site-packages/salt/client/__init__.py", line 920, in get_iter_returns
    ret = {raw['data']['id']: raw['data']['data']}
KeyError: 'id'
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 349, in salt_main
    client.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/salt.py", line 195, in run
    for full_ret in cmd_func(**kwargs):
  File "/usr/lib/python2.6/site-packages/salt/client/__init__.py", line 595, in cmd_cli
    **kwargs):
  File "/usr/lib/python2.6/site-packages/salt/client/__init__.py", line 1297, in get_cli_event_returns
    expect_minions=(verbose or show_timeout)
  File "/usr/lib/python2.6/site-packages/salt/client/__init__.py", line 920, in get_iter_returns
    ret = {raw['data']['id']: raw['data']['data']}
KeyError: 'id'
[root@flt-master /]#
```

Digging around on the minion logs I saw:
```
2015-08-05 17:05:57,034 [salt.utils.event                         ][DEBUG   ][1047] Sending event - data = {'_stamp': '2015-08-05T17:05:57.034351', 'pretag': None, 'cmd': '_minion_event', 'tag': '_salt_error', 'data': {'message': 'The minion function caused an exception', 'args': ['The minion function caused an exception'], '_stamp': '2015-08-05T17:05:57.030000'}, 'id': 'pod-minion-aus-jsf-admin'}
```

**Comment by jfindlay** [comment]
@yannis666, thanks for finding this.

**Comment by cachedout** [comment]
Fix in #26078

**Comment by cachedout** [comment]
Merged #26078. Closing this issue.
